src/ApollonEditor.ts

```diff
@@ -0,0 +1,137 @@
+import { createElement, RefObject, createRef } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { Store } from 'redux';
+import { Application } from './scenes/Application';
+import { State } from './components/Store';
+import { DiagramType } from './domain/Diagram';
+import { ElementKind } from './domain/Element';
+import { RelationshipKind } from './domain/Relationship';
+import { Styles } from './components/Theme';
+
+export interface UMLModel {
+  type: DiagramType;
+  interactive: Selection;
+  elements: { [id: string]: UMLElement };
+  relationships: { [id: string]: UMLRelationship };
+}
+
+export { DiagramType };
+export { ElementKind };
+export { RelationshipKind };
+export { Styles };
+
+export interface UMLElement {
+  id: string;
+  name: string;
+  owner: string | null;
+  type: ElementKind;
+  bounds: { x: number; y: number; width: number; height: number };
+}
+
+export const enum Location {
+  North,
```

Comment:
while this is a really nice relation to Compass in Artemis ;-) I would suggest that we use Up, Left, Down, Right here instead. Also we might want to call it Direction instead of Location. What do you think?

Reply:
Sounds good! Applied rename to:
```
export const enum Direction {
  Up,
  Right,
  Down,
  Left,
}
```